test/e2e/gameserver_test.go

```diff
@@ -769,6 +769,22 @@ func TestGameServerTcpUdpProtocol(t *testing.T) {
 	assert.Equal(t, "ACK TCP: Hello World !\n", replyTCP)
 }
 
+func TestGameServerWithoutPort(t *testing.T) {
+	t.Parallel()
+	gs := framework.DefaultGameServer(framework.Namespace)
+	gs.Spec.Ports = nil
+
+	_, valid := gs.Validate()
+	assert.True(t, valid)
+
+	readyGs, err := framework.CreateGameServerAndWaitUntilReadyWithOptionalPortCheck(framework.Namespace, gs, false)
+	if err != nil {
+		assert.FailNow(t, "Could not get a GameServer ready", err.Error())
+	}
+
+	assert.Nil(t, readyGs.Spec.Ports)
```

Comment:
Just a thought. This might be better, just because nil vs an empty array are both valid return results. Not a strong opinion.

https://pkg.go.dev/github.com/stretchr/testify/assert#Empty

```suggestion
	assert.Empty(t, readyGs.Spec.Ports) 
```

Reply:
Yes, you are right. assert.Empty itself does a null check.

Done